Client/Frontend/TabBar/SearchViewController.swift

```diff
@@ -0,0 +1,87 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import UIKit
+
+private let ReuseIdentifier = "cell"
+
+protocol SearchViewControllerDelegate: class {
+    func didClickSearchResult(url: NSURL?)
+}
+
+class SearchViewController: UIViewController {
+    weak var delegate: SearchViewControllerDelegate?
+    private var searchEngines = [OpenSearchEngine]()
+
+    private var searchText = ""
+    private var tableView = UITableView()
+
+    required init(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+    }
+
+    override init() {
+        // The empty initializer of UIViewController creates the class twice (?!),
+        // so override it here to avoid calling it.
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    override func viewDidLoad() {
+        tableView.dataSource = self
+        tableView.delegate = self
+        tableView.registerClass(SearchTableViewCell.self, forCellReuseIdentifier: ReuseIdentifier)
+
+        // Make the row separators span the width of the entire table.
+        tableView.layoutMargins = UIEdgeInsetsZero
+        tableView.separatorInset = UIEdgeInsetsZero
+
+        view.addSubview(tableView)
+        tableView.snp_makeConstraints { make in
+            make.edges.equalTo(self.view)
+            return
+        }
+    }
+
+    func setSearchEngines(engines: [OpenSearchEngine]) {
```

Comment:
Why a setter? Isn't that non-idiomatic for Swift? I think `searchViewController.searchEngines = mySearchEngines` is preferred?